default.nix

```diff
@@ -1,66 +1,26 @@
-{ frontend ? false }:
+{ ... } @ args:
+
 let
-  obelisk-src = (import <nixpkgs> {}).fetchFromGitHub {
-    owner = "obsidiansystems";
-    repo = "obelisk";
-    rev = "6c38599615eddba1b9f8dfb845f7404f53ed8053";
-    sha256 = "0pqhppn2cb69v7r6wbg2zrx5ylxbgq7bl7qilarfbmyd9gcph9h4";
-  };
-  reflex-platform = (import obelisk-src {}).reflex-platform;
 
-  gargoyle-src = reflex-platform.nixpkgs.fetchFromGitHub {
-    owner = "obsidiansystems";
-    repo = "gargoyle";
-    rev = "2c19c569325ad76694526e9b688ccdbf148df980";
-    sha256 = "0257p0qd8xx900ngghkjbmjnvn7pjv05g0jm5kkrm4p6alrlhfyl";
-  };
-  groundhog-src = reflex-platform.nixpkgs.fetchFromGitHub {
-    owner = "obsidiansystems";
-    repo = "groundhog";
-    rev = "c2f18be45e3233f6268c8468eb0732dd6b2e8009";
-    sha256 = "1r9i78bsnm6idbvp87gjklnr10g7c83nsbnrffkyrn1wmd7zzqdn";
-  };
-in reflex-platform.project ({ pkgs, ... }: {
-  packages = {
-    rhyolite-backend = ./backend;
-    rhyolite-backend-snap = ./backend-snap;
-    rhyolite-common = ./common;
-    rhyolite-frontend = ./frontend;
-    rhyolite-datastructures = ./datastructures;
-    rhyolite-aeson-orphans  = ./aeson-orphans;
+  obelisk = import ./.obelisk/impl args;
+  reflex-platform = obelisk.reflex-platform;
+  pkgs = obelisk.nixpkgs;
+  inherit (pkgs) lib;
+
+  # Use can use these manually if you don’t want to use rhyolite.project.
+  # It will be needed if you need it to combine with multiple overrides.
+  haskellOverrides = import ./overlay.nix { inherit pkgs; };
```

Comment:
The overlay shouldn't close over things like that. Instead you can use this lolzy trick to get `pkgs` from within the Haskell overlay: https://github.com/obsidiansystems/obelisk/blob/master/default.nix#L55 .